cli/ops/timers.rs

```diff
@@ -1,57 +1,198 @@
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
 use super::dispatch_json::{Deserialize, JsonOp, Value};
+use crate::deno_error::bad_resource;
+use crate::futures::Stream;
 use crate::ops::json_op;
+use crate::resources;
+use crate::resources::CoreResource;
 use crate::state::ThreadSafeState;
 use deno::*;
 use futures::Future;
+use futures::Poll;
 use std;
 use std::time::Duration;
 use std::time::Instant;
+use tokio::timer::Delay;
+use tokio::timer::Interval;
 
 pub fn init(i: &mut Isolate, s: &ThreadSafeState) {
+  i.register_op("now", s.core_op(json_op(s.stateful_op(op_now))));
   i.register_op(
-    "global_timer_stop",
-    s.core_op(json_op(s.stateful_op(op_global_timer_stop))),
+    "set_interval",
+    s.core_op(json_op(s.stateful_op(op_set_interval))),
   );
   i.register_op(
-    "global_timer",
-    s.core_op(json_op(s.stateful_op(op_global_timer))),
+    "poll_interval",
+    s.core_op(json_op(s.stateful_op(op_poll_interval))),
+  );
+  i.register_op(
+    "clear_interval",
+    s.core_op(json_op(s.stateful_op(op_clear_interval))),
+  );
+  i.register_op(
+    "set_timeout",
+    s.core_op(json_op(s.stateful_op(op_set_timeout))),
+  );
+  i.register_op(
+    "poll_timeout",
+    s.core_op(json_op(s.stateful_op(op_poll_timeout))),
+  );
+  i.register_op(
+    "clear_timeout",
+    s.core_op(json_op(s.stateful_op(op_clear_timeout))),
   );
-  i.register_op("now", s.core_op(json_op(s.stateful_op(op_now))));
 }
 
-fn op_global_timer_stop(
-  state: &ThreadSafeState,
-  _args: Value,
+struct IntervalResource {
+  interval: Interval,
+}
```

Comment:
Maybe this is something I missed in the resource refactor, but why does Interval need to be wrapped in this IntervalResource? This seems like unnecessary indirection/boilerplate. Is it only for the "inspect_repr" method?

Reply:
It needs to implement `CoreResource` trait so it's downcastable to concrete type when accessing the resource. I wanted to implement this trait directly for `Interval` but Rust doesn't allow to do that.